src/CacheService.php

```diff
@@ -107,7 +107,7 @@ protected function recursiveGet(string $key, int $level = 0): array
 
         [$value, $keyFound] = $this->recursiveGet($key, $level + 1);
 
-        if ($keyFound || (!$keyFound && $adapter->cacheNotFoundKeys())) {
```

Comment:
This seems redundant to me: `(!$keyFound && $adapter->cacheNotFoundKeys())` since `$keyFound` will pass the condition. `$keyFound || $adapter->cacheNotFoundKeys()` will do, right?